helm/idol-library/templates/_ingress.tpl

```diff
@@ -81,6 +87,21 @@ spec:
 {{- end -}}
 {{- end }}
```

Comment:
I think you can move the additional secret generation logic here rather than putting it in each chart
```
{{- end -}}
{{ if $ingress.tls.secretName -}}
---
{{ include "idol-library.ingress.secret" (dict "tls" $ingress.tls) -}}
{{- end }}
```

Reply:
This doesn't work, seems like the helm template can only create a single resource. I'll try and find another way though.